File: backend/src/PageNest.Domain/Entities/Payment.cs
using PageNest.Domain.Enums;

namespace PageNest.Domain.Entities;

public class Payment
{
    public Guid Id { get; set; }
    public Guid OrderId { get; set; }
    public Order Order { get; set; }
    public decimal Amount { get; set; }
    public string StripePaymentIntentId { get; set; }
    public PaymentStatus Status { get; set; }
}

File: frontend/pagenest.client/src/types/payment.ts
import type { PaymentStatus } from '../utils/dictionary';

export interface IPayment {
  id?: string;
  orderId: string;
  amount: number | string;
  stripePaymentIntentId: string;
    status: PaymentStatus;
    createdAt?: Date;
}

export interface IPaymentRowProps {
    payment: IPayment;
}
